packages/core/src/interfaces/components.ts

```diff
@@ -26,7 +27,7 @@ export interface LayoutComponentChild {
 /**
  * customize the legend component
  */
-export interface LegendOptions {
```

Comment:
we should probably just include `truncation` inside `LegendOptions` pointing to `TruncationOptions` like so:
```ts
{
  truncation?: TruncationOptions;
}
```

just so we don't extend various interfaces in the future...

Reply:
I tried the code above but it didn't seem to work. For some reason it will not recognize the truncation in the interface. However it works if I extends TruncationOptions..